internal/provisioner/README.md

```diff
@@ -1,6 +1,60 @@
-# provisioner
+# Provisioners
 
 The following subdirectories contain concrete Provisioner implementations.
 
 - [plain](plain/README.md) - provisions `plain+v0` k8s bundles
 - [registry](registry/README.md) - provisions `registry+v1` OLM bundles
+
+## Global Provisioner Concepts
+Below is an example of the provisioner reconciliation flow:
+
+```mermaid
+graph TD
+    C[Provisioner]
+    C -->|Watches| D[Bundle]
+    C -->|Watches| E[BundleInstance]
+    D -->|References| F[Content]
+    E -->|Creates/Manages| F[Content]
+```
+### Make bundle content available but do not install it
+
+There is a natural separation between sourcing of the content and application of that content via two separate RukPak
+APIs, `Bundle` and `BundleInstance`. A user can specify a particular `Bundle` to be available in the cluster for
+inspection before any application of the resources. Given a `Bundle` resource named `my-bundle`, the plain provisioner
```

Comment:
nit: since this is the top-level provisioner README, should we have references to the plain provisioner specifically here? I would expect we could treat this doc as more generic, and defer to the individual provisioner README docs for specifics.

Reply:
The reference to plain here should definitely not be there. Fixing all of the reference to it in this top-level document.